packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -275,12 +277,14 @@ export function getBeaconBlockApi({
                 if (isOptimisticBlock(canonical)) {
                   executionOptimistic = true;
                 }
+                finalized = false;
```

Comment:
the finalized block could return in `chain.forkChoice.getCanonicalBlockAtSlot(block.message.slot)`

Reply:
you're right, let's add a comment for this place to note that